frontend-mobile/lib/widgets/coffee_call.dart

```diff
@@ -1,55 +1,62 @@
 import 'package:coffee_caller/communication/coffee_caller_protocol.dart';
 import 'package:coffee_caller/communication/socket_client.dart';
-import 'package:coffee_caller/view/cubit/coffee_call_cubit.dart';
-import 'package:coffee_caller/view/cubit/coffee_call_state.dart';
-import 'package:coffee_caller/view/settings_screen.dart';
+import 'package:coffee_caller/storage/settings_storage.dart';
+import 'package:coffee_caller/widgets/no_username.dart';
+import 'package:coffee_caller/widgets/cubit/coffee_call_cubit.dart';
+import 'package:coffee_caller/widgets/cubit/coffee_call_state.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 
-class CoffeeCallScreen extends StatelessWidget {
-  const CoffeeCallScreen({
+class CoffeeCall extends StatelessWidget {
+  const CoffeeCall({
     super.key,
   });
 
   @override
   Widget build(BuildContext context) {
-    return Scaffold(
-      appBar: AppBar(
-        title: const Text('Coffee Caller'),
-        actions: [
-          TextButton.icon(
-            style: TextButton.styleFrom(foregroundColor: Colors.white),
-            onPressed: () {
-              Navigator.push(
-                context,
-                MaterialPageRoute(builder: (context) => const SettingsScreen()),
-              );
-            },
-            icon: const Icon(Icons.settings),
-            label: const Text('Settings'),
-          ),
-        ],
-      ),
-      body: BlocProvider(
-        create: (_) => CoffeeCallCubit(
-          socket: context.read<SocketClient>(),
-          protocol: CoffeeCallerProtocol(
-            socketClient: context.read<SocketClient>(),
-          ),
-        )
-          ..init()
-          ..connect(),
-        child: const CoffeeCallBody(),
-      ),
+    return BlocProvider(
+      create: (_) => CoffeeCallCubit(
+        socket: context.read<SocketClient>(),
+        protocol: CoffeeCallerProtocol(
+          socketClient: context.read<SocketClient>(),
+        ),
+      )
+        ..init()
+        ..connect(),
+      child: const CoffeeCallBody(),
     );
   }
 }
 
-class CoffeeCallBody extends StatelessWidget {
+class CoffeeCallBody extends StatefulWidget {
   const CoffeeCallBody({super.key});
 
+  @override
+  State<StatefulWidget> createState() => _CoffeeCallBodyState();
+}
+
+class _CoffeeCallBodyState extends State<CoffeeCallBody> {
+  late String userName = '';
+
+  @override
+  void initState() {
+    super.initState();
+    _setUsername();
+  }
+
+  void _setUsername() {
+    getUsername().then((value) {
+      setState(() {
+        userName = value;
+      });
+    });
+  }
+
   @override
   Widget build(BuildContext context) {
+    if (userName == '') {
+      return NoUsername(onUsernameSet: _setUsername);
```

Comment:
Dunno is this is the "flutter way" to work with callbacks

Reply:
The more low-level your widget the more is it appropriate to use callbacks.